spring-cloud-gcp-vision/src/main/java/org/springframework/cloud/gcp/vision/DocumentOcrResultSet.java

```diff
@@ -0,0 +1,198 @@
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.vision.v1.TextAnnotation;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+/**
+ * Represents the parsed OCR content for an document in the provided range of pages.
+ *
+ * @author Daniel Zou
+ */
+public class DocumentOcrResultSet {
+
+	private static final Pattern OUTPUT_PAGE_PATTERN = Pattern.compile("output-(\\d+)-to-(\\d+)\\.json");
+
+	private final TreeMap<Integer, OcrPageRange> ocrPageRanges;
+
+	private final int minPage;
+
+	private final int maxPage;
+
+	DocumentOcrResultSet(Collection<Blob> pages) {
+		this.ocrPageRanges = new TreeMap<>();
+
+		for (Blob blob : pages) {
+			OcrPageRange pageRange = extractPageRange(blob);
+			ocrPageRanges.put(pageRange.startPage, pageRange);
+		}
+
+		this.minPage = this.ocrPageRanges.values().stream()
+				.mapToInt(range -> range.startPage)
+				.min()
+				.getAsInt();
+
+		this.maxPage = this.ocrPageRanges.values().stream()
+				.mapToInt(range -> range.endPage)
+				.max()
+				.getAsInt();
+	}
+
+	/**
+	 * Returns the minimum page number in the result set.
+	 *
+	 * @return the lowest page number in the result set.
+	 */
+	public int getMinPage() {
+		return this.minPage;
+	}
+
+	/**
+	 * Returns the maximum page number in the result set.
+	 *
+	 * @return the highest page number in the result set.
+	 */
+	public int getMaxPage() {
+		return this.maxPage;
+	}
+
+
+	/**
+	 * Retrieves the parsed OCR information of the page at index {@code pageNumber} of the
+	 * document. The page number must be a value between {@link DocumentOcrResultSet#getMinPage()}
+	 * and {@link DocumentOcrResultSet#getMaxPage()}.
+	 *
+	 * <p>
+	 * This returns a TextAnnotation object which is Google Cloud Vision's representation of a
+	 * page of a document. For more information on reading this object, see:
+	 * https://cloud.google.com/vision/docs/reference/rpc/google.cloud.vision.v1#google.cloud.vision.v1.TextAnnotation
+	 *
+	 * @param pageNumber the page number of the document
+	 * @return the {@link TextAnnotation} representing the page of the document
+	 * @throws InvalidProtocolBufferException if the OCR information for the page failed to be
+	 *     parsed
+	 *
+	 */
+	public TextAnnotation getPage(int pageNumber) throws InvalidProtocolBufferException {
+		if (pageNumber < this.minPage || pageNumber > this.maxPage) {
+			throw new IndexOutOfBoundsException("Page number out of bounds: " + pageNumber);
+		}
+
+		Map.Entry<Integer, OcrPageRange> floorEntry = this.ocrPageRanges.floorEntry(pageNumber);
+		OcrPageRange pageRange = floorEntry.getValue();
+		List<TextAnnotation> documentPages = DocumentOcrTemplate.parseJsonBlob(pageRange.blob);
```

Comment:
I think we discussed caching the parsed json blobs. Can we do that?

Reply:
Done.